engine/src/main/java/org/terasology/telemetry/Metrics.java

```diff
@@ -27,101 +28,50 @@
 import org.terasology.telemetry.metrics.MonsterKilledMetric;
 import org.terasology.telemetry.metrics.SystemContextMetric;
 
-import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 
 /**
  * Metrics class is similar to {@link org.terasology.config.Config}, it stores the telemetry information.
  * Once a new metric is used, the new metric instance should be added in this class to show the metric value in ui.
  */
+@API
 public class Metrics {
 
-    private static final Logger logger = LoggerFactory.getLogger(Metrics.class);
-
-    private SystemContextMetric systemContextMetric;
-
-    private ModulesMetric modulesMetric;
-
-    private GameConfigurationMetric gameConfigurationMetric;
-
-    private BlockDestroyedMetric blockDestroyedMetric;
-
-    private BlockPlacedMetric blockPlacedMetric;
-
-    private GamePlayMetric gamePlayMetric;
-
-    private MonsterKilledMetric monsterKilledMetric;
-
-    public Metrics() {
-
-    }
+    private Map<String, Metric> metricsMap = new HashMap<>();
```

Comment:
This maps classnames to metrics, how about a name like `metricByClassName` or `classNameToMetric`? `metricsMap` is very generic and adds little extra information to the field definition.